### DISCUSSION TITLE
Run a live reload without connecting to the device to the PC

### DISCUSSION THREAD

**Discussion by Sampath-Lokuge** [ionic-team/capacitor#3194] [Q&A] [answered]
Hi, 

Do you know how to run a live reload without connecting to the device to the PC? 

`ionic cap run android -l --external` This is working fine if I'll connect my device to PC or with Emulator. But what about without connection to the PC? I can remember it was there on the Ionic/Cordova era.

**Comment by imhoffd** [comment, marked as answer]
## iOS

1. Connect your device to your computer with USB to begin setup.
1. In Xcode, navigate to Window -> Devices and Simulators. Find your phone under Devices and make sure "Connect via network" is checked (device and computer need to be on the same Wi-Fi network).

## Android

### Enabling TCP Mode

1. Connect your device to your computer with USB to begin setup. Run `adb devices` to make sure the device is listed. 
1. Switch ADB from USB to TCP mode.

    ```
    adb tcpip 5555
    ```

1. Establish a connection from your computer to the device by using the device's IP on the shared LAN (device and computer need to be on the same Wi-Fi network). You can get the device's IP by going to Settings -> About phone -> Network -> IP address.

    ```
    adb connect 192.168.0.xxx
    ```

1. Unplug the phone and run `adb devices` to ensure the connection was established (the device will now be listed as an IP).

### Enabling USB Mode

To switch back to USB mode, run `adb usb`.

### Troubleshooting

If adb connection is lost, try running `adb connect <ip>` again. If that doesn't work, run `adb kill-server` and start over.

There's [a section in the Android docs](https://developer.android.com/studio/command-line/adb#wireless) if you'd like to read more.

### Bonus 🤫 

You can use `localhost` in the Web View even with live reload in USB or TCP mode by forwarding the port of the dev server from device to computer (usually 8100):

```
adb reverse tcp:8100 tcp:8100
```

**Comment by Sampath-Lokuge** [comment]
Hi @dwieeb, Could you tell me how can I do this with iOS also? I have a problem with iOS since I do not have a USB-C to USB adapter on Macbook Air.

**Reply by imhoffd** [reply]
Sure, I updated my answer. You still need a USB cable to setup the wireless connection, unfortunately.

**Reply by Sampath-Lokuge** [reply]
Yes, That is the case. Anyway, the good news is I have purchased the original Apple USB C to USB Adapter and now I do not have any issues. Thanks for the great feedback though.
